### PR TITLE
Fix links in the UserGuide and Enable Assertions on the Team Repo

### LINKED ISSUE #69: Fix User Guide ToC links and Enable assertions in the team repo
Also enables Assertions on our repo by editing the build gradle file as specified [here](https://se-education.org/guides/tutorials/gradle.html#enabling-assertions), as required by the end of Section 3 on [this](https://nus-cs2103-ay2324s1.github.io/website/schedule/week10/project.html#3-start-the-next-iteration) page

### PR DESCRIPTION
**Pull request by ghost**
Fixed the broken links in the UG and added a placeholder Glossary section to be worked on so that the TOC link works.

Also enables Assertions on our repo by editing the build gradle file as specified [Here](https://se-education.org/guides/tutorials/gradle.html#enabling-assertions), as required by the end of Section 3 on [this](https://nus-cs2103-ay2324s1.github.io/website/schedule/week10/project.html#3-start-the-next-iteration) page

Closes #69 
Contributes to #49

### CODE REVIEW

**Review by jingjie88 — APPROVED** [review]
LGTM!

File: build.gradle [hunk 1]
```diff
@@ -69,4 +69,8 @@ shadowJar {
     archiveFileName = 'addressbook.jar'
 }
 
+run {
+    enableAssertions = true
```
Comment (kiwibang) [inline build.gradle:73]: Great idea configuring Gradle to enable assertions instead of using the IntelliJ VM options 👍

File: docs/UserGuide.md [hunk 2]
```diff
@@ -5,17 +5,17 @@ title: User Guide
...
-  - [Interview Management Command](##InterviewManagementCommand)
-- [Glossary](#Glossary)
+- [Introduction](#introduction)
+- [Quick Start](#quick-start)
+- [GUI Summary](#gui-summary)
+- [Features](#features)
+- [Frequently Asked Questions](#frequently-asked-questions)
+- [Known Issues](#known-issues)
+- [Command Summary](#command-summary)
+  - [General Command](#general-command)
```
Comment (kiwibang) [inline docs/UserGuide.md:15]: Thanks for updating the TOC for the UG 👍

**Review by kiwibang — APPROVED** [review]
LGTM! I see no issues!

**Pull request merged**